### DISCUSSION TITLE
Question: Can I customize the nginx-fpm port?

### DISCUSSION THREAD

**Discussion by Sammyjo20** [serversideup/docker-php#352] [Q&A]
Hey,

I've finally had the chance to check out serversideup/php and I absolutely love it. Great work on v3 as well! 👍

I'm currently attempting to build a system for my work which will allow us to spin up separate staging deployments per PR/branch we have and I'm going to try using Docker for it (and this image). For now, I'm going to try having a separate nginx container which acts like a proxy and forwards traffic to the serversideup/php-fpm-nginx images, each running on their own ports. 

I'm currently struggling to find if I can customise the default port (8080) that is used to something different, so we can use one network but multiple ports so my nginx proxy can forward traffic based on URL. 

Any help would be greatly appreciated and if what I'm doing is stupid, please do say as I'm still quite new to Docker and this image! 😅

**Comment by Sammyjo20** [comment]
To add to this, I'm also trying not to expose any ports to the host machine except for port 80/443 of my proxy.

**Comment by Sammyjo20** [comment]
Please disregard this! I worked out how to use a custom network, assign static IPs and I can proxy through to each of the individual containers running port 8080. Thank you! 🙏

**Reply by jaydrogers** [reply]
Thanks for the compliments!!

Sorry I couldn't get back to you sooner, I was deep in code 🤪

Stay tuned for some upcoming updates with Spin. It might dramatically help you get off the ground with Docker with the new updates I am working on.

One of them is making it easier to use a proxy like Traefik.

Rather than configuring static IP addresses, everything is routed based off the domain and a label. Oh and it comes with zero-downtime deployments too 🎉

https://github.com/serversideup/spin-template-laravel/blob/56b3f0eaedf9f7d14393a126f167da380a73a7e6/template/docker-compose.dev.yml#L27-L33
